### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by sachinnagpal**

### CODE REVIEW

File: src/ui/script.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  let updateServiceStatus = document.getElementById("update-service-status")
+  if (status.stoppingServices) {
+    updateServiceSpinner.classList.add("fa", "fa-solid", "fa-spinner", "fa-spin")
+    updateServiceStatus.innerHTML = `Stopping Mark Services`
+  }
+  if (Object.keys(status).includes(`stoppingServices`)) {
+    if (!status.stoppingServices) {
+      if (Object.keys(status).includes('stopped-successfully')) {
+        let isSuccessfullyStopped = status['stopped-successfully']
+        if (isSuccessfullyStopped) {
```
Comment (riturajborpujari) [inline src/ui/script.js:49]: use this naming convention for boolean variables like `isUpdatedSuccessfully` above

File: src/ui/script.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+ipcRenderer.on('update:status', (status) => {
+  console.log(JSON.stringify(status))
+  let updateStatus = document.getElementById("update-status")
+  if(Object.keys(status).includes(`updating`)) {
+    if(status.updating) {
+      // updateServiceSpinner.classList.add("fa", "fa-solid", "fa-spinner", "fa-spin")
+      updateStatus.innerHTML = 'Updating Mark Services'
+    }else {
+      // updateServiceSpinner.classList.remove("fa", "fa-solid", "fa-spinner", "fa-spin")
+      let updatedSuccesfully = status['updated-successfully']
```
Comment (riturajborpujari) [inline src/ui/script.js:29]: Use name `isUpdatedSuccessfully`

File: src/ui/script.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      }
+    }
+})
+
+
+ipcRenderer.on('service:status', (status) => {
+  serviceStatuses.push(status)
+  let serviceStatus = document.getElementById("service-status")
+  let serviceSpinner = document.getElementById("service-spinner")
+  if (serviceStatuses.length > 0 && serviceStatuses.length < 7) {
```
Comment (riturajborpujari) [inline src/ui/script.js:136]: Why is this `< 7 ` check required?

if this has a required logic behind, extract away into its method and give a relevant name

File: src/ui/script.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    serviceStatus.innerHTML = 'Runing up the services'
+    serviceSpinner.classList.add("fa", "fa-solid", "fa-spinner", "fa-spin")
+  } else if (serviceStatuses.length === 7) {
```
Comment (riturajborpujari) [inline src/ui/script.js:139]: Same comment as above.

Extract out into something like `areAllServiceRunning` if required

File: src/ui/script.js [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+      serviceSpinner.classList.add("fa", "fa-check-square-o")
+      let serviceStatus = document.getElementById("service-already-installed-status")
+      serviceStatus.innerHTML = 'Mark is up and running'
+    } else if (installationStatus.startingServices) {
+      let serviceSpinner = document.getElementById("service-already-installed-spinner")
+      serviceSpinner.classList.add("fa", "fa-solid", "fa-spinner", "fa-spin")
+      let serviceStatus = document.getElementById("service-already-installed-status")
+      serviceStatus.innerHTML = `Starting up the services`
+    }
+    else if (installationStatus.servicesNotRunning.length > 0) {
```
Comment (riturajborpujari) [inline src/ui/script.js:178]: if status fields `listedServicesRunning`, `startingServices`, and `servicesDownloaded` are mutually exclusive, they should be the possible values of a single `serviceStatus` field

```ts
export type serviceStatus: "listedServicesRunning" | "startingServices" | "servicesDownloaded"
```

**Review by riturajborpujari — CHANGES_REQUESTED** [review]

File: src/backend/update.js [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+const setMainWindowUpdate = function (window) {
+  mainWindow = window
+}
+const updateAndStartMark = async function () {
+  log.info(`Sending alert for update`)
+  utils.sendAlert('update:status', {
+    updating: true
+  })
+  try {
+    stopServicesAndAlert()
```
[the remark this answers is not in the document]
Comment (sachinnagpal) [reply src/backend/update.js:21]: Moved it to a new service file itself.

File: src/backend/update.js [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+    utils.sendAlert('update:status', {
+      installing: false,
+      'installation-successful': true
+    })
+  } catch (err) {
+    log.info(`Installed failed `)
+    log.info(err)
+    utils.sendAlert('update:status', {
+      installing: false,
+      'installation-successful': false
```
[the remark this answers is not in the document]
Comment (sachinnagpal) [reply src/backend/update.js:148]: For installer case we have that, but for update not doing it currently. Can bring it up in next phase

File: src/ui/script.js (hunk 3 above)
Comment (sachinnagpal) [reply src/ui/script.js:109]: Will try to simplify this complete logic in next phase. Will discuss it with @shailendraO4S as well to right it in more cleaner way.

**Review by riturajborpujari — APPROVED** [review]

**Comment by sachinnagpal** [comment]
There are few comments which need to be resolved in phase 2. Mostly from front end and one related to init method in backend. 
cc: @riturajborpujari

**Pull request merged**